afterpay/src/main/kotlin/com/afterpay/android/internal/AfterpayInstalment.kt

```diff
@@ -19,34 +20,59 @@ internal sealed class AfterpayInstalment {
+                currencyLocales.count() == 1 -> currencyLocales.first()
+                currencyLocales.contains(configuration.locale) -> configuration.locale
+                else -> Locales.validSet.first { Currency.getInstance(it) == configuration.currency }
+            }
+
+            val localCurrency = Currency.getInstance(clientLocale)
             val currencySymbol = configuration.currency.getSymbol(currencyLocale)
-            val usCurrencySymbol = Currency.getInstance(Locales.US).getSymbol(Locales.US)
-            val localCurrency = Currency.getInstance(configuration.locale)
 
-            val currencyFormatter = (NumberFormat.getCurrencyInstance(currencyLocale) as DecimalFormat).apply {
+            val usCurrencySymbol = Currency.getInstance(Locales.EN_US).getSymbol(Locales.EN_US)
+            val gbCurrencySymbol = Currency.getInstance(Locales.EN_GB).getSymbol(Locales.EN_GB)
+            val euCurrencySymbol = Currency.getInstance(Locales.FR_FR).getSymbol(Locales.FR_FR)
+
+            val currencyFormatter = (NumberFormat.getCurrencyInstance(clientLocale) as DecimalFormat).apply {
                 this.currency = configuration.currency
             }
 
-            if (configuration.locale == Locales.US) {
+            if (clientLocale == Locales.EN_US) {
                 currencyFormatter.apply {
+                    when (currencySymbol) {
+                        euCurrencySymbol -> this.applyPattern("#,##0.00¤")
+                    }
+
                     decimalFormatSymbols = decimalFormatSymbols.apply {
                         this.currencySymbol = when (configuration.currency) {
-                            Currency.getInstance(Locales.AUSTRALIA) -> "A$"
-                            Currency.getInstance(Locales.NEW_ZEALAND) -> "NZ$"
-                            Currency.getInstance(Locales.CANADA) -> "CA$"
+                            Currency.getInstance(Locales.EN_AU) -> "A$"
+                            Currency.getInstance(Locales.EN_NZ) -> "NZ$"
+                            Currency.getInstance(Locales.EN_CA) -> "CA$"
+                            Currency.getInstance(Locales.FR_CA) -> "CA$"
                             else -> currencySymbol
                         }
                     }
                 }
-            } else if (currencySymbol == usCurrencySymbol && configuration.currency != localCurrency) {
+            } else if (configuration.currency != localCurrency) {
                 currencyFormatter.apply {
-                    this.applyPattern("¤#,##0.00 ¤¤")
+                    decimalFormatSymbols = decimalFormatSymbols.apply {
+                        this.currencySymbol = currencySymbol
+                    }
+
+                    when (currencySymbol) {
+                        usCurrencySymbol -> this.applyPattern("¤#,##0.00 ¤¤")
+                        gbCurrencySymbol -> this.applyPattern("¤#,##0.00")
+                        euCurrencySymbol -> this.applyPattern("#,##0.00¤")
```

Comment:
The format for EUR and European locales uses comma (`,`) for the decimal separator and period (`.`) for the thousands separator. I'm not sure if all combinations of the EUR currency and the European locales are covered correctly in this section. In the JS library for web, `fr-CA` uses the `#.##0,00` format for all currencies, `es-ES`, `fr-FR` and `it-IT` use it only for EUR, and the other locales do not use it at all.

CC @luvgangwani92

Reply:
Looks like my question is answered in [afterpay/src/test/kotlin/com/afterpay/android/AfterpayInstalmentTest.kt](https://github.com/afterpay/sdk-android/pull/151/files#diff-3da6acadd4d58953ada822ab6d2692679840feec05c0dea0735fc5af923d7df4). Only difference is that here, the `#.##0,00` format is used for all currencies by the `es-ES`, `fr-FR` and `it-IT` locales. This seems to make more sense at first glance, but maybe @luvgangwani92 has some further context to share.

Reply:
Not sure how kotlin reads those patterns, but looking solely at the positions of the `,` and `.` characters I agree with @bfirth. Also, @ScottAntonacAP is there a way we can think of replicating all the price cases that we have for the JS lib in other SDK's? This might increase our confidence in the amount parsing and presentation. We might need to make the [price cases](https://github.com/AfterpayTouch/ei-lib-js/blob/main/src/assets/price-cases.js) available as an independent JSON. This need not be a part of this PR though.

Reply:
@benfterpay regarding the formatting the above is only the pattern. The decimal place symbol in this method of patterning is always a dot `.` and the separator a comma `,`. These are displayed correctly based on the consumers locale (as you pointed out in the test cases).

@luvgangwani92 yes agreed that we should have a higher number of test cases. Have raised an internal ticket for this.